RavenDB-24623 - AI Agent: small TOODs and leftovers

shaharhikri opened the pull request:
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-24623/AI-Agent-small-TOODs-and-leftovers

### Additional description
create sharded EP and throw,
add XML to RunConversationOperation,
make second message after summarization (in SummarizeAsync) role 'assistant', 
remove where we can the CreateSampleDataOperation from tests, go over most of the test and use the client api

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [x] New feature

### How risky is the change?

- [ ] Low 
- [ ] Moderate 
- [ ] High
- [x] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [x] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing
- [x] Existing tests verify the correct behavior

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

karmeli87 commented on src/Raven.Server/Documents/Handlers/AI/Agents/ShardedAiAgentHandler.cs:
```diff
@@ -0,0 +1,41 @@
+        [RavenShardedAction("/databases/*/admin/ai/agent", "PUT")]
+        public Task AddOrModifyAiAgent()
+        {
+            throw new NotSupportedException("This operation is not available from a specific shard");
```
better message and use `NotSupportedInShardingException`

shaharhikri replied:
Done

karmeli87 replied:
you didn't changed the message.
let's write something like `AI Agents for a sharded database are currently not supported`

shaharhikri replied:
Done

karmeli87 commented on test/SlowTests/Server/Documents/AI/AiAgent/AiAgentBasics.cs:
```diff
@@ -118,70 +123,20 @@ public async Task CanResumeConversation(Options options, GenAiConfiguration conf
-
-        [RavenTheory(RavenTestCategory.Ai)]
-        [RavenGenAiData(IntegrationType = RavenAiIntegration.OpenAi, DatabaseMode = RavenDatabaseMode.Single, CheckCanConnect = false, NightlyBuildRequired = false)]
-        public async Task AnswerActionToolRequest(Options options, GenAiConfiguration config)
```
where did that go?

shaharhikri replied:
After changing it, it turned out to be identical to the test on `AiAgentClientApiBasics`..

shaharhikri committed:
RavenDB-24623 - create sharded EP and throw,
add XML to RunConversationOperation,
make second message after summarization (in SummarizeAsync) role 'assistant',
remove where we can the CreateSampleDataOperation from tests,
go over most of the test and use the client api

shaharhikri commented:
Failing test isn't related to the changes
